File: Planetary-Formation-main/Storage/PlanarScalarData.java
package Storage;

import Storage.Positionals.Vector2;

public interface PlanarScalarData {
    // gets the value at a point on the field
    public double getPoint(Vector2 point);
    // Mutates every point on the field by setting it to the value output by the operation
    public void mutateLocal(ScalarQuadTree.LocalMutator operation);
    // Returns one quarter of the field, scaled up
    // false means 0, true means 1
    public PlanarScalarData getQuadrant(boolean x, boolean y);
}

File: Planetary-Formation-main/storage/PlanarScalarData.kt
package storage

import storage.positionals.Vector2
import storage.positionals.Vector3
import storage.STL.TriangleFace
import storage.ScalarQuadTree.LocalMutator
import java.util.function.Function

interface PlanarScalarData {
    fun getPoint(point: Vector2): Long
    fun mutateLocal(operation: ScalarQuadTree.LocalMutator)
    fun evaluateLocal(operation: ScalarQuadTree.PointEvaluator)
    fun getQuadrant(x: Boolean, y: Boolean): PlanarScalarData
    fun getValues(): LongArray
    //fun exportTriangles(projector: Function<Vector3, Vector3>): Array<TriangleFace>
}